[Tan Zhang-Hng Ervin] iP

MakiseKurisuX opened the pull request:
### Helios Chatbot 

> **"Helios Chatbot is here to make task management easier!"**  

## Features  
- Manage **todos, deadlines, and events** 
- Supports **add, delete, mark, find** commands 
- Persistent **storage system**
- Simple **CLI-based interface** 

And it is **FREE!**

## Commands  
- `todo <task>` → Adds a **todo**  
- `deadline <task> /by <date-time>` → Adds a **deadline**  
- `event <task> /from <start> /to <end>` → Adds an **event**  
- `mark <task #>` / `unmark <task #>` → ✅ / ❌ Task status  
- `delete <task #>` → 🗑 Delete a task  
- `list` → 📋 Show all tasks  
- `find <keyword>` → 🔍 Search tasks  
- `bye` → 👋 Exit  

## Task List  
- [x] Managing Tasks
- [x] Managing Deadlines 
- [ ] GUI version (future update)  

## Example Usage  
```sh
$ todo Finish PR  
$ deadline Submit assignment /by 02/15/2009 23:59  
$ list

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

MakiseKurisuX committed:
Add ChatBot.java

Create ChatBot.java
----------------------------
Level 0 Task for Week 2 iP

MakiseKurisuX committed:
Update ChatBot.java

Update ChatBot.java with Echo
----------------------------
Level 1 Task for Week 2 iP

MakiseKurisuX committed:
Update ChatBot.java

Update ChatBot.java with Add, List
----------------------------
Level 2 Task for Week 2 iP

MakiseKurisuX committed:
Update ChatBot.java

Update ChatBot.java with Marking
----------------------------
Level 3 Task for Week 2 iP

MakiseKurisuX committed:
Update ChatBot.java, add new files

Update ChatBot.java with ToDos, Events, Deadlines

Improved on User Inputs, using startsWith instead of using split to find first word.

Created new classes for Event, Task, ToDo.
----------------------------
Level 4 Task for Week 2 iP

MakiseKurisuX committed:
Update ChatBot.java, add Exception Handling

Update ChatBot.java to use HeliosException for Exception Handling.

Fix unnoticed error from previous Level 4 commit. [extra instance of status icon showing]
----------------------------
Level 5 Task for Week 2 iP

MakiseKurisuX committed:
Update ChatBot.java

Update ChatBot.java with addition of Delete Function.
----------------------------
Level 6 Task for Week 2 iP

MakiseKurisuX committed:
Update ChatBot.java

Update ChatBot.java with Save Function.

Add Getters into Class Files.
----------------------------
Level 7 Task for Week 3 iP

MakiseKurisuX committed:
Add DateTimeParser.java

Create new DateTimeParser for chatbot to better understand time.

Modify Getters as necessary.
----------------------------
Level 8 Task for Week 3 iP

MakiseKurisuX committed:
Add OOP, Update ChatBot.java

Make use of OOP Style, adding new Classes and Objects
----------------------------
A-OOP Task for Week 3 iP

MakiseKurisuX committed:
Repackage Files

Repackage files, divide classes into multiple packages
----------------------------
A-Package Task for Week 3 iP

MakiseKurisuX committed:
Merge remote-tracking branch 'origin/add-gradle-support'

MakiseKurisuX committed:
Add JUnit Tests

Add 2 JUnit Tests
----------------------------
A-JUnit Task for Week 3 iP

MakiseKurisuX committed:
Update Gradle Build File

Update Gradle Build File to package into JAR file.
----------------------------
A-Jar Task for Week 3 iP

MakiseKurisuX committed:
Add JavaDoc to Files

Add JavaDoc to relevant Java files.
----------------------------
A-JavaDoc Task for Week 3 iP

MakiseKurisuX committed:
Update ChatBot, Parser, TaskList

Update ChatBot, Parser and TaskList with new function for finding tasks
----------------------------
level-9 Task for Week 3 iP

MakiseKurisuX committed:
Fix error

Fix error with regards to find function.

MakiseKurisuX committed:
Update Files for Coding Standard

Updating Files to go closer in line with CS2103 Coding Standards
----------------------------
A-CodingStandard Task for Week 3 iP

MakiseKurisuX committed:
Fix Errors

Fix Errors from Updating Coding Standard

MakiseKurisuX committed:
Update TextUiTesting

Add TextUiTesting Tests for Chatbot
----------------------------------------
A-TextUiTesting for Week 2 iP Task

twhjames commented on src/main/java/chatbot/ChatBot.java:
```diff
@@ -0,0 +1,136 @@
+  /*
+   * Runs the chatbot, handles user input and executes commands in a loop.
+   * The chatbot will run commands until the user types 'bye'.
+   */
```
I like your descriptive header comments for each public class and methods 👍

twhjames commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,92 @@
+      throw new HeliosException("Deadline must have a /by time.");
+    }
+    return parts;
+  }
```
This is a well-named method as it conveys both its purpose and the plural nature of the returned array

twhjames commented on src/main/java/chatbot/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+ * Manages an ArrayList of tasks, allows you to add, remove, mark and list tasks.
+ */
+public class TaskList {
+  private ArrayList<Task> tasks;
```
Well-named variable as it clearly indicates that it stores a list of tasks

twhjames commented on src/main/java/chatbot/Ui.java:
```diff
@@ -0,0 +1,67 @@
+    return scanner.nextLine();
+  }
+
+  /*
+   * Displays welcome message.
+   */
+  public void welcomeMessage() {
+    System.out.println("____________________________________________________________\r\n" + //
+            " Hello! I'm Ervin Chatbot!\r\n" + //
+            " What can I do for you?\r\n" + //
+            "____________________________________________________________\r\n");
+  }
```
Consider abstracting this print statement into a separate method such as `printDivider()` for better readability and reusability.

```suggestion
  /*
   * Displays welcome message.
   */
  public void welcomeMessage() {
    printDivider();
    System.out.println(" Hello! I'm Ervin Chatbot!\r\n What can I do for you?\r\n");
    printDivider();
  }
```
```java
private void printDivider() {
    System.out.println("____________________________________________________________\r\n"

}
```

twhjames commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,57 @@
+ * Inherits from the Task Class.
+ */
+public class Event extends Task {
+    private LocalDateTime  from;
+    private LocalDateTime  to;
```
Consider removing the additional whitespace

```suggestion
    private LocalDateTime from;
    private LocalDateTime to;
```

MakiseKurisuX committed:
Fix Tabulation

Fix Tabulation to modify to recommended size - 4 spaces

MakiseKurisuX committed:
Add JavaFX files

Add JavaFX Files for GUI application
----------------------------------------
A-Level-10 for Week 4 iP Task

AerollingOverDeepBlueSeas commented on src/main/java/chatbot/Ui.java:
```diff
@@ -0,0 +1,67 @@
+     * Displays welcome message.
+     */
+    public void welcomeMessage() {
+        System.out.println("____________________________________________________________\r\n" + //
```
It's really minor (just indentation), but the indentation level the strings in this method seems to be different from the other methods in the same file.

Btw, good work on the formatting overall. It's really easy to read! Aside from that, I don't have much nitpicks.

AerollingOverDeepBlueSeas commented on src/main/java/chatbot/ChatBot.java:
```diff
@@ -0,0 +1,137 @@
+                }
+            } else if (task.startsWith("unmark")) {
+                try {
+                            int taskIndex = parser.getIndex(task);
```
Unindent this line slightly?

AerollingOverDeepBlueSeas commented on src/main/java/chatbot/ChatBot.java:
```diff
@@ -0,0 +1,137 @@
+package chatbot;
+
+import java.util.Scanner;
+import java.util.ArrayList;
```
Are these imports necessary for this file?

AerollingOverDeepBlueSeas commented on src/main/java/chatbot/DateTimeParser.java:
```diff
@@ -0,0 +1,67 @@
+ * Utility class for parsing and formatting date and time values.
+ */
+public class DateTimeParser {
+
```
I like how two different representations of date/datetime are allowed!

MakiseKurisuX committed:
Update for JavaFX

Update and Add files to make use of JavaFX

Created ChatController Class in accordance to OOP principles
----------------------------------------
A-Level-10 for Week 4 iP Task

MakiseKurisuX committed:
Update ChatBot, TaskList, Object Files

Update ChatBot, TaskList and relevant Object Files to Allow for Sorting of Tasks
----------------------------
C-Sort Task for Week 5 iP

MakiseKurisuX committed:
Fix Scrolling, Text View

Chatbot Window: Does not support scrolling

Users are unable to view previous messages once they overflow
beyond visible window.

Replace VBox with a ScrollPane that wraps the VBox, allowing
users to scroll manually while still displaying new messages
properly.

ScrollPane provides built-in scrolling functionality, ensuring
that users can freely navigate past messages while maintaining
smooth auto-scrolling for new messages.

-----------------------------------------------

DialogBox - Messages with long content were getting cut off

Long messages were not fully displayed, making it difficult
to read the entire content.

Set text.setMaxWidth(Double.MAX_VALUE) in DialogBox to allow
messages to fully expand

Using Double.MAX_VALUE ensures that messages dynamically adjust
to available space rather than being forcibly constrained

MakiseKurisuX committed:
Update ChatBot, DialogBox

Update ChatBot, DialogBox to add Assertions
----------------------------
A-Assertions Task for Week 5 iP

MakiseKurisuX committed:
Update Parser.java, Storage.java

Removed printing output from Storage.java

Removed Magic Numbers / Strings from
Storage.java

MakiseKurisuX committed:
Update ChatBot.java, Ui.java

Implement SLAP to extract code blocks
from ChatBot.java

MakiseKurisuX committed:
Update ChatBot.java

Update ChatBot.java to remove magic literals
and improve variable names

MakiseKurisuX committed:
Update TaskList.java

Update TaskList to improve general
Code Quality, implement SLAP practices

MakiseKurisuX committed:
Fix Errors

Fix Errors related to HeliosException

MakiseKurisuX committed:
Update ChatController.java

Update ChatController to remove magic
variables.

MakiseKurisuX committed:
Update DateTimeParser.java

Update DateTimeParser for specificity
Fix Javadoc Error

MakiseKurisuX committed:
Update DialogBox.java

Remove Magic Numbers from DialogBox

MakiseKurisuX committed:
Merge branch 'branch-A-CodeQuality'

MakiseKurisuX committed:
Update ChatBot.java

Fix Error from PR Merger.

MakiseKurisuX committed:
Update ChatController.java

Update Welcome Message

MakiseKurisuX committed:
Update README.md

Update README to add user guide.

MakiseKurisuX committed:
Add Ui.png, fix error

Add Ui.png

Fix error with building of Jar file.

MakiseKurisuX committed:
Update Storage.java

Add Method to Create Directory
if it does not exist.

MakiseKurisuX committed:
Update build.gradle, chatbot.fxml

Update build.gradle, chatbot.fxml for release

MakiseKurisuX committed:
Update README

Update README

MakiseKurisuX committed:
Update Parser, README

Fix Issue from earlier Bug Testing

MakiseKurisuX committed:
Update ChatController.java

Fix "Enter" button not working on
click

Close application when "bye" is
typed

MakiseKurisuX committed:
Update Error Handling

Update Error Handling for Exceptions.

MakiseKurisuX committed:
Update Parser.java

Add additional error handling.

MakiseKurisuX committed:
Update Event.java

Add Error Handling for Event Timing

MakiseKurisuX committed:
Update JUnit Tests

Update removed methods,
add exceptions where needed
